### ISSUE TITLE
Separate 'perl' version requirement metadata

### ISSUE THREAD

**Issue by Grinnz** [Perl-Toolchain-Gang/CPAN-Meta#103]
'perl' is not a normal prerequisite, so it should have its own metadata field separate from 'prereqs'. Namely, 1. it should never be installed by the CPAN installer, so it can only cause installation to fail, and following from that 2. suggests/recommends for 'perl' makes no sense in the context of the CPAN installer. Having the field be a regular Version Range as it is in the prereqs object would be sensible.

**Comment by karenetheridge** [comment]
I think this is a good candidate for metaspec v3.

**Comment by Leont** [comment]
I don't think we can do anything with it on the runtime-configure-build-test axis, but I do think it could be useful on the requires-recommends-suggests axis, if only to communicate to the end-user.

**Comment by Grinnz** [comment]
I would consider documentation a more than sufficient place to communicate a recommended perl version; I cannot think of a tool that would benefit from having that information in metadata.

**Comment by karenetheridge** [comment]
> I don't think we can do anything with it on the runtime-configure-build-test axis

The cpan client could inspect that before running Makefile.PL, so as to disqualify a distribution for install earlier; presently it doesn't find out until it runs Makefile.PL and hits a `use 5.024;` statement.

**Comment by Grinnz** [comment]
But runtime/configure/build/test is definitely not a useful distinction for required perl version, forgot to mention that in the original comment.

**Comment by karenetheridge** [comment]
This could be prototyped as:

```
    x_perl => {
        requires => '5.010',
        recommends => '5.012',
        suggests => '5.022',
        conflicts => '>= 5.024',  # just for completeness - but this would mean e.g. "definitely doesn't work there as there are breaking changes in XS"
    }
```

**Comment by Leont** [comment]
What practical problem would be solved by this? How would this change make software simpler at this stage?

**Comment by rjbs** [comment]
That's exactly my question, too.  Handling "perl" in the prereqs is a solved problem.

**Comment by haarg** [comment]
I don't think there is any real gain by trying to split perl out at this point.  Having perl included in the standard prereq section may not be entirely consistent, but it is well established and is handled by everything already.  I can contrive situations where a test recommends on perl would make sense, and I don't see the harm in allowing things like test requires being different from runtime requires.
